lib/register.ts

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright 2016-2020 Balena Ltd.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import * as randomstring from 'randomstring';
```

Comment:
Do we really need to use a dependency here?
https://nodejs.org/api/crypto.html#crypto_crypto_randombytes_size_callback

I know it's part of the original code, but it could be nice to handle it here, given the large overhaul this PR does.

Reply:
@CameronDiver I'd rather not mess with that as I'd like it to stay consistent with anywhere else we might generate a key, and also since (if done correctly) it's a patch level change there's no need to bundle it into a major bump